packages/ckeditor5-ui/src/menubar/utils.ts

```diff
@@ -169,13 +169,25 @@ export const MenuBarBehaviors = {
 	 * Tracks the keyboard focus interaction on the menu bar view. It is used to determine if the nested items
 	 * of the menu bar should render focus rings after first interaction with the keyboard.
 	 */
-	trackKeyboardFocusInteraction( menuBarView: MenuBarView ): void {
+	enableFocusHighlightOnInteraction( menuBarView: MenuBarView ): void {
 		let isKeyPressed: boolean = false;
 
-		menuBarView.on<ObservableChangeEvent<boolean>>( 'change:isOpen', () => {
-			menuBarView.hadKeyboardFocusInteraction = false;
+		menuBarView.on<ObservableChangeEvent<boolean>>( 'change:isOpen', ( _, evt, isOpen ) => {
+			if ( !isOpen ) {
+				menuBarView.isFocusBorderEnabled = false;
+
+				// Reset the flag when the menu bar is closed, menu items tend to intercept `keyup` event
+				// and sometimes, after pressing `enter` on focused item, `isKeyPressed` stuck in `true` state.
+				isKeyPressed = false;
+			}
 		} );
 
+		menuBarView.listenTo( menuBarView.element!, 'click', () => {
+			if ( menuBarView.isOpen && menuBarView.element!.matches( ':focus-within' ) ) {
+				menuBarView.isFocusBorderEnabled = true;
+			}
+		}, { useCapture: true } );
+
```

Comment:
This is incorrect. It reacts to every mouse click. E.g. I can click "Format" and then click "Text >" and it starts showing focus. On the other hand, it fixes the main issue a bit coincidentally. We shouldn't need to listen to a mouse click it all. After all, this is about keyboard interactions.

Instead Alt+F9 should cause `isFocusBorderEnabled = true` and all further iteractions should simply work. I guess that the problem is that when Alt+F9 happens, the focus is not in the menu, so listener below (`keydown`) does not trigger.

Let's summarize what works correctly now, and what needs to be revisited:

1.  When I mouse click a menu, it doesn't get a blue border anymore: correct.
2.  When I used keyboard to navigate menu (e.g. arrow down, enter), menu gets a blue border and it follows mouse hovers everywhere, including top-level menu: correct.
3.  When I Alt+F9 and click on the same top-level menu ("File"), it gets a blue border and border and focus follows mouse hover: correct but with incorrect fix. Also if I click another menu (e.g. "View") it works correctly.
4.  But when I Alt+F9 and start hovering other top-level menus, without clicking on them, the border does not move. What's more interesting, the focus does not move either! E.g. if I Alt+F9 and then hover a different item (e.g. "Insert") and press arrow right, the focus goes to "Edit" instead of "Format". This is different than e.g. when I click on the menu item (e.g. "Insert"), and then hover, then focus changes, and arrow keys work correctly. I don't know internals, so I don't know why there's a difference between Alt+F9 and click.

So, to summarize again:

1.  Alt+F9 should trigger `isFocusBorderEnabled`.
2.  Alt+F9 should causes focus to follow mouse.

Reply:
@scofalik 

> This is incorrect. It reacts to every mouse click. E.g. I can click "Format" and then click "Text >" and it starts showing focus. On the other hand, it fixes the main issue a bit coincidentally. We shouldn't need to listen to a mouse click it all. After all, this is about keyboard interactions.

It fixes edge case. If you open submenu using click it focuses first item of newly opened menu. It handles situation when user click button down while not having any item highlighted in menu.

Reply:
@scofalik Alt+f9 is already handled and it should behave as you describe. I'll check that again but you have to have editor focused to make any keystrokes work

Reply:
You are correct, I checked it on Windows VM and something is broken. I'm checking.

Reply:
@scofalik Fixed, please reopen if it still occurs.